modules/exploits/linux/ftp/proftp_telnet_iac.rb

```diff
@@ -277,29 +277,29 @@ def check
     status = CheckCode::Safe
     bannerArray = banner.split('.')
 
-    if bannerArray.count() > 0 and !bannerArray[3].nil?
-     # gets 1 char on the third part of version number.
-     relnum = bannerArray[2][0..0]
-     tmp = bannerArray[2].split(' ')
-     # gets extra string info of version number.
-     # example: 1.2.3rc ('rc' string)
-     extra = tmp[0][1..(tmp[0].length - 1)]
-     if relnum == '2'
-      if extra.length > 0
-       if extra[0..1] == 'rc'
-        v = extra[2..extra.length].to_i
-        if v and v > 2
-         status = CheckCode::Appears
+    if bannerArray.count() > 0 && !bannerArray[3].nil?
+      # gets 1 char on the third part of version number.
+      relnum = bannerArray[2][0..0]
+      tmp = bannerArray[2].split(' ')
+      # gets extra string info of version number.
+      # example: 1.2.3rc ('rc' string)
+      extra = tmp[0][1..(tmp[0].length - 1)]
+      if relnum == '2'
+        if extra.length > 0
+          if extra[0..1] == 'rc'
+            v = extra[2..extra.length].to_i
+            if v && v > 2
+              status = CheckCode::Appears
+            end
+          else
+            status = CheckCode::Appears
+          end
+        end
+      elsif relnum == '3'
+        if [ '', 'a', 'b', 'c' ].include?(extra)
```

Comment:
Is this not meant to include the character `r` here as well given the original code? Will leave this to Spencer to do the rest of his review on but thought I'd point this out as it seems the original code was also checking for the character `r`.

Reply:
No, the rc versions aren't affected so they should return the status of "safe".